### PR TITLE
feat: sdk logger typescript typings

### PR DESCRIPTION
**Pull request by eoln**
- clean up typescript typings
- added typings for Logger module
- update jest and @types/node dev-deps
- volunteer to become code owner

### CODE REVIEW

**Review by msk- — DISMISSED** [review]
Take this as tentative approval of the logger typings based on my limited experience of ts. Probably best to get another approval before merging. (Although, OTOH, as this is predominantly anciliary to the functionality, maybe it's not so urgent.)

**Review by kleyow — DISMISSED** [review]
lgtm

File: src/index.d.ts
Comment (lewisdaly) [inline src/index.d.ts:517]: I think this should have a return signature, which I think will just be `LoggerStringify` like you have above

File: src/index.d.ts
Comment (lewisdaly) [inline src/index.d.ts:549]: This should also have `opts` like in the constructor, with no defaults

**Review by lewisdaly — CHANGES_REQUESTED** [review]

File: src/index.d.ts
Comment (eoln) [reply src/index.d.ts]: nice catch @lewisdaly I always can count on your eyeballs resolution and focus :-) Fixed

**Review by lewisdaly — APPROVED** [review]
Looks good to me!

**Pull request merged**
